File: src/main/java/com/matag/game/turn/phases/PhaseUtils.java
package com.matag.game.turn.phases;

import static com.matag.game.turn.phases.beginning.UntapPhase.UT;
import static com.matag.game.turn.phases.ending.CleanupPhase.CL;
import static com.matag.game.turn.phases.main1.Main1Phase.M1;
import static com.matag.game.turn.phases.main2.Main2Phase.M2;

public class PhaseUtils {
  public static boolean isMainPhase(String phase) {
    return phase.equals(M1) || phase.equals(M2);
  }

  public static boolean isPriorityAllowed(String phase) {
    return !(phase.equals(UT) || phase.equals(CL));
  }
}

File: src/main/kotlin/com/matag/game/turn/phases/PhaseUtils.kt
package com.matag.game.turn.phases

import com.matag.game.turn.phases.beginning.UntapPhase
import com.matag.game.turn.phases.ending.CleanupPhase
import com.matag.game.turn.phases.main1.Main1Phase
import com.matag.game.turn.phases.main2.Main2Phase

object PhaseUtils {
    fun isMainPhase(phase: String): Boolean {
        return phase == Main1Phase.M1 || phase == Main2Phase.M2
    }

    fun isPriorityAllowed(phase: String): Boolean {
        return !(phase == UntapPhase.UT || phase == CleanupPhase.CL)
    }
}
